### PR TITLE
Added `extend`/`extend_unchecked` for `MutableBooleanArray`

### PR DESCRIPTION
**Pull request by VasanthakumarV**
This PR adds `TrustedLen`-based extend and extend_unchecked methods for `MutableBooleanArray`.

I have mostly followed the implementation for [`MutablePrimitiveArray`](https://github.com/jorgecarleitao/arrow2/blob/main/src/array/primitive/mutable.rs).

### CODE REVIEW

File: src/array/boolean/mutable.rs [hunk 1]
```diff
@@ -104,6 +104,69 @@ impl MutableBooleanArray {
...
     }
 
+    /// Extends the [`MutableBooleanArray`] from an iterator of values of trusted len.
+    /// This differs from `extend_trusted_len` which accepts in iterator of optional values.
+    #[inline]
+    pub fn extend_trusted_len_values<I>(&mut self, iterator: I)
+    where
+        I: TrustedLen<Item = bool>,
+    {
+        unsafe { self.extend_trusted_len_values_unchecked(iterator) }
```
Comment (jorgecarleitao) [inline src/array/boolean/mutable.rs:114]:
```suggestion
        /// Safety: `I` is `TrustedLen`
        unsafe { self.extend_trusted_len_values_unchecked(iterator) }
```

File: src/array/boolean/mutable.rs [hunk 2]
```diff
@@ -240,33 +297,53 @@ impl MutableBooleanArray {
...
         } else {
-            validity.push(false);
-            false
+            validity.push_unchecked(false);
+            bool::default()
         };
-        values.push(item);
+        values.push_unchecked(item);
     }
-    assert_eq!(
```
Comment (jorgecarleitao) [inline src/array/boolean/mutable.rs:264]: Could we keep the `debug_assert_eq!` here?

File: tests/it/array/boolean/mutable.rs [hunk 3]
```diff
@@ -64,3 +64,35 @@ fn reserve() {
...
+    assert_eq!(a.validity(), None);
+    assert_eq!(a.values(), &MutableBitmap::from([true, true, false]));
+
+    let mut a = MutableBooleanArray::new();
+    a.push(None);
+    a.extend_trusted_len_values(vec![true, false].into_iter());
+    assert_eq!(
+        a.validity(),
+        Some(&MutableBitmap::from([false, true, true]))
+    );
```
Comment (jorgecarleitao) [inline tests/it/array/boolean/mutable.rs:97]:
```suggestion
    assert_eq!(
        a.validity(),
        Some(&MutableBitmap::from([false, true, true]))
    );
    assert_eq!(
        a.values(),
        &MutableBitmap::from([false, true, false])
    );
```

**Review by jorgecarleitao — APPROVED** [review]
As usual, beautiful. Thank you very much!

Left some suggestions (the `Safety` should be applied to every usage of `unsafe`), but otherwise ready to go.

File: tests/it/array/boolean/mutable.rs (hunk 3 above)
Comment (VasanthakumarV) [reply tests/it/array/boolean/mutable.rs:97]: Thanks for pointing this out, the missing assertion has been added

File: src/array/boolean/mutable.rs (hunk 2 above)
Comment (VasanthakumarV) [reply src/array/boolean/mutable.rs:264]: @jorgecarleitao I have added `debug_assert_eq`, but I had to create a variable to store the length value before expansion and use it later in the assert statement

File: src/array/boolean/mutable.rs (hunk 1 above)
Comment (VasanthakumarV) [reply src/array/boolean/mutable.rs:115]: The Safety comment was added to every place inside the file where we use unsafe blocks, but `///` wasn't possible

**Pull request merged**